This notebook will be used for the Capstone Project

In [19]:
import pandas as pd
import numpy as np

In [20]:
print('Hello Capstone Project Course!')

Hello Capstone Project Course!


In [21]:
table = pd.read_html("https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M")
table

[    Postal Code           Borough  \
 0           M1A      Not assigned   
 1           M2A      Not assigned   
 2           M3A        North York   
 3           M4A        North York   
 4           M5A  Downtown Toronto   
 ..          ...               ...   
 175         M5Z      Not assigned   
 176         M6Z      Not assigned   
 177         M7Z      Not assigned   
 178         M8Z         Etobicoke   
 179         M9Z      Not assigned   
 
                                          Neighbourhood  
 0                                         Not assigned  
 1                                         Not assigned  
 2                                            Parkwoods  
 3                                     Victoria Village  
 4                            Regent Park, Harbourfront  
 ..                                                 ...  
 175                                       Not assigned  
 176                                       Not assigned  
 177                

In [22]:
column_names = ['Postal Code','Borough','Neighbourhood', 'Latitude', 'Longitude'] 

# instantiate the dataframe
neighborhoods = pd.DataFrame(columns=column_names)
neighborhoods

Empty DataFrame
Columns: [Postal Code, Borough, Neighbourhood, Latitude, Longitude]
Index: []

In [23]:
neighborhoods_data = table[0]
neighborhoods_data

Postal Code           Borough  \
0           M1A      Not assigned   
1           M2A      Not assigned   
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
..          ...               ...   
175         M5Z      Not assigned   
176         M6Z      Not assigned   
177         M7Z      Not assigned   
178         M8Z         Etobicoke   
179         M9Z      Not assigned   

                                         Neighbourhood  
0                                         Not assigned  
1                                         Not assigned  
2                                            Parkwoods  
3                                     Victoria Village  
4                            Regent Park, Harbourfront  
..                                                 ...  
175                                       Not assigned  
176                                       Not assigned  
177                                       Not assigned  
178  Mimico NW, The Queensway West, South of Bloor,...  
179                                       Not assigned  

[180 rows x 3 columns]

In [24]:
neighborhoods = pd.DataFrame(data=neighborhoods_data)
neighborhoods


Postal Code           Borough  \
0           M1A      Not assigned   
1           M2A      Not assigned   
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
..          ...               ...   
175         M5Z      Not assigned   
176         M6Z      Not assigned   
177         M7Z      Not assigned   
178         M8Z         Etobicoke   
179         M9Z      Not assigned   

                                         Neighbourhood  
0                                         Not assigned  
1                                         Not assigned  
2                                            Parkwoods  
3                                     Victoria Village  
4                            Regent Park, Harbourfront  
..                                                 ...  
175                                       Not assigned  
176                                       Not assigned  
177                                       Not assigned  
178  Mimico NW, The Queensway West, South of Bloor,...  
179                                       Not assigned  

[180 rows x 3 columns]

In [25]:
cleaned_neighborhoods = neighborhoods[neighborhoods.Borough != 'Not assigned']
cleaned_neighborhoods = cleaned_neighborhoods.reset_index(drop=True)
cleaned_neighborhoods.head(15)


Postal Code           Borough  \
0          M3A        North York   
1          M4A        North York   
2          M5A  Downtown Toronto   
3          M6A        North York   
4          M7A  Downtown Toronto   
5          M9A         Etobicoke   
6          M1B       Scarborough   
7          M3B        North York   
8          M4B         East York   
9          M5B  Downtown Toronto   
10         M6B        North York   
11         M9B         Etobicoke   
12         M1C       Scarborough   
13         M3C        North York   
14         M4C         East York   

                                        Neighbourhood  
0                                           Parkwoods  
1                                    Victoria Village  
2                           Regent Park, Harbourfront  
3                    Lawrence Manor, Lawrence Heights  
4         Queen's Park, Ontario Provincial Government  
5             Islington Avenue, Humber Valley Village  
6                                      Malvern, Rouge  
7                                           Don Mills  
8                     Parkview Hill, Woodbine Gardens  
9                            Garden District, Ryerson  
10                                          Glencairn  
11  West Deane Park, Princess Gardens, Martin Grov...  
12             Rouge Hill, Port Union, Highland Creek  
13                                          Don Mills  
14                                   Woodbine Heights

In [26]:
cleaned_neighborhoods = cleaned_neighborhoods.sort_values(by=['Postal Code'])
cleaned_neighborhoods = cleaned_neighborhoods.reset_index(drop=True)
cleaned_neighborhoods.head(15)

Postal Code      Borough                                      Neighbourhood
0          M1B  Scarborough                                     Malvern, Rouge
1          M1C  Scarborough             Rouge Hill, Port Union, Highland Creek
2          M1E  Scarborough                  Guildwood, Morningside, West Hill
3          M1G  Scarborough                                             Woburn
4          M1H  Scarborough                                          Cedarbrae
5          M1J  Scarborough                                Scarborough Village
6          M1K  Scarborough        Kennedy Park, Ionview, East Birchmount Park
7          M1L  Scarborough                    Golden Mile, Clairlea, Oakridge
8          M1M  Scarborough    Cliffside, Cliffcrest, Scarborough Village West
9          M1N  Scarborough                        Birch Cliff, Cliffside West
10         M1P  Scarborough  Dorset Park, Wexford Heights, Scarborough Town...
11         M1R  Scarborough                                  Wexford, Maryvale
12         M1S  Scarborough                                          Agincourt
13         M1T  Scarborough            Clarks Corners, Tam O'Shanter, Sullivan
14         M1V  Scarborough  Milliken, Agincourt North, Steeles East, L'Amo...

In [27]:
cleaned_neighborhoods.shape

(103, 3)

In [34]:

import types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

# @hidden_cell
# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.
# You might want to remove those credentials before you share the notebook.
client_3023e3c906e24611aa5dd4943d051e4a = ibm_boto3.client(service_name='s3',
    ibm_api_key_id='eX_7YKcZNHdpizUI3k2YcnBmyA9qaCY31xHC2_BGh4C9',
    ibm_auth_endpoint="https://iam.cloud.ibm.com/oidc/token",
    config=Config(signature_version='oauth'),
    endpoint_url='https://s3-api.us-geo.objectstorage.service.networklayer.com')

body = client_3023e3c906e24611aa5dd4943d051e4a.get_object(Bucket='datasciencecapstone-donotdelete-pr-xmhato38mrgzsz',Key='longlat.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

df_data_1 = pd.read_csv(body)
df_data_1.head(10)


Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476
5         M1J  43.744734 -79.239476
6         M1K  43.727929 -79.262029
7         M1L  43.711112 -79.284577
8         M1M  43.716316 -79.239476
9         M1N  43.692657 -79.264848

In [38]:
cleaned_neighborhoods2 = cleaned_neighborhoods.merge(df_data_1)
cleaned_neighborhoods2

Postal Code      Borough  \
0           M1B  Scarborough   
1           M1C  Scarborough   
2           M1E  Scarborough   
3           M1G  Scarborough   
4           M1H  Scarborough   
..          ...          ...   
98          M9N         York   
99          M9P    Etobicoke   
100         M9R    Etobicoke   
101         M9V    Etobicoke   
102         M9W    Etobicoke   

                                         Neighbourhood   Latitude  Longitude  
0                                       Malvern, Rouge  43.806686 -79.194353  
1               Rouge Hill, Port Union, Highland Creek  43.784535 -79.160497  
2                    Guildwood, Morningside, West Hill  43.763573 -79.188711  
3                                               Woburn  43.770992 -79.216917  
4                                            Cedarbrae  43.773136 -79.239476  
..                                                 ...        ...        ...  
98                                              Weston  43.706876 -79.518188  
99                                           Westmount  43.696319 -79.532242  
100  Kingsview Village, St. Phillips, Martin Grove ...  43.688905 -79.554724  
101  South Steeles, Silverstone, Humbergate, Jamest...  43.739416 -79.588437  
102                Northwest, West Humber - Clairville  43.706748 -79.594054  

[103 rows x 5 columns]

Above we have our final list of neighborhoods with latitude and longitude.  

In [39]:
CLIENT_ID = 'WBWWOIBFNM0MIRB5XE3CWH34X3G11OFZJT1U0HH4MTLPY05T' # your Foursquare ID
CLIENT_SECRET = 'HJRKMZP4WAYLUMXLU1QYYL3ZNFVVFJY5OWYAUHL1TTP40C22' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version
LIMIT = 100 # A default Foursquare API limit value

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: WBWWOIBFNM0MIRB5XE3CWH34X3G11OFZJT1U0HH4MTLPY05T
CLIENT_SECRET:HJRKMZP4WAYLUMXLU1QYYL3ZNFVVFJY5OWYAUHL1TTP40C22


In [41]:
import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [43]:
toronto_venues = getNearbyVenues(names=cleaned_neighborhoods2['Neighbourhood'],
                                   latitudes=cleaned_neighborhoods2['Latitude'],
                                   longitudes=cleaned_neighborhoods2['Longitude']
                                  )

Malvern, Rouge
Rouge Hill, Port Union, Highland Creek
Guildwood, Morningside, West Hill
Woburn
Cedarbrae
Scarborough Village
Kennedy Park, Ionview, East Birchmount Park
Golden Mile, Clairlea, Oakridge
Cliffside, Cliffcrest, Scarborough Village West
Birch Cliff, Cliffside West
Dorset Park, Wexford Heights, Scarborough Town Centre
Wexford, Maryvale
Agincourt
Clarks Corners, Tam O'Shanter, Sullivan
Milliken, Agincourt North, Steeles East, L'Amoreaux East
Steeles West, L'Amoreaux West
Upper Rouge
Hillcrest Village
Fairview, Henry Farm, Oriole
Bayview Village
York Mills, Silver Hills
Willowdale, Newtonbrook
Willowdale, Willowdale East
York Mills West
Willowdale, Willowdale West
Parkwoods
Don Mills
Don Mills
Bathurst Manor, Wilson Heights, Downsview North
Northwood Park, York University
Downsview
Downsview
Downsview
Downsview
Victoria Village
Parkview Hill, Woodbine Gardens
Woodbine Heights
The Beaches
Leaside
Thorncliffe Park
East Toronto, Broadview North (Old East York)
The Danforth West, 

In [44]:
toronto_venues.shape

(2117, 7)

Now we have a list of 2117 venues in Toronto that we can group by neighborhood.

In [45]:
toronto_venues.head(20)

Neighborhood  Neighborhood Latitude  \
0                           Malvern, Rouge              43.806686   
1                           Malvern, Rouge              43.806686   
2   Rouge Hill, Port Union, Highland Creek              43.784535   
3   Rouge Hill, Port Union, Highland Creek              43.784535   
4        Guildwood, Morningside, West Hill              43.763573   
5        Guildwood, Morningside, West Hill              43.763573   
6        Guildwood, Morningside, West Hill              43.763573   
7        Guildwood, Morningside, West Hill              43.763573   
8        Guildwood, Morningside, West Hill              43.763573   
9        Guildwood, Morningside, West Hill              43.763573   
10       Guildwood, Morningside, West Hill              43.763573   
11       Guildwood, Morningside, West Hill              43.763573   
12                                  Woburn              43.770992   
13                                  Woburn              43.770992   
14                                  Woburn              43.770992   
15                                  Woburn              43.770992   
16                               Cedarbrae              43.773136   
17                               Cedarbrae              43.773136   
18                               Cedarbrae              43.773136   
19                               Cedarbrae              43.773136   

    Neighborhood Longitude                                              Venue  \
0               -79.194353                                            Wendy’s   
1               -79.194353                              Interprovincial Group   
2               -79.160497                              Royal Canadian Legion   
3               -79.160497  SEBS Engineering Inc. (Sustainable Energy and ...   
4               -79.188711                                     RBC Royal Bank   
5               -79.188711                                  G & G Electronics   
6               -79.188711                                         Sail Sushi   
7               -79.188711                                   Big Bite Burrito   
8               -79.188711                              Enterprise Rent-A-Car   
9               -79.188711                              Woburn Medical Centre   
10              -79.188711                       Lawrence Ave E & Kingston Rd   
11              -79.188711                                           Eggsmart   
12              -79.216917                                          Starbucks   
13              -79.216917                                        Tim Hortons   
14              -79.216917                                 Korean Grill House   
15              -79.216917     Toronto Ali Sami Yen Spor Kompleksi (Lionhill)   
16              -79.239476                                Federick Restaurant   
17              -79.239476                           Drupati's Roti & Doubles   
18              -79.239476                                        Thai One On   
19              -79.239476                       Centennial Recreation Centre   

    Venue Latitude  Venue Longitude              Venue Category  
0        43.807448       -79.199056        Fast Food Restaurant  
1        43.805630       -79.200378                  Print Shop  
2        43.782533       -79.163085                         Bar  
3        43.782371       -79.156820  Construction & Landscaping  
4        43.766790       -79.191151                        Bank  
5        43.765309       -79.191537           Electronics Store  
6        43.765951       -79.191275                  Restaurant  
7        43.766299       -79.190720          Mexican Restaurant  
8        43.764076       -79.193406         Rental Car Location  
9        43.766631       -79.192286              Medical Center  
10       43.767704       -79.189490                Intersection  
11       43.767800       -79.190466              Breakfast Spot  
12       43.770037  

In [46]:
toronto_venues.groupby('Neighborhood').count()

Neighborhood Latitude  \
Neighborhood                                                             
Agincourt                                                            4   
Alderwood, Long Branch                                               7   
Bathurst Manor, Wilson Heights, Downsview North                     23   
Bayview Village                                                      4   
Bedford Park, Lawrence Manor East                                   24   
...                                                                ...   
Willowdale, Willowdale East                                         34   
Willowdale, Willowdale West                                          6   
Woburn                                                               4   
Woodbine Heights                                                     7   
York Mills West                                                      2   

                                                 Neighborhood Longitude  \
Neighborhood                                                              
Agincourt                                                             4   
Alderwood, Long Branch                                                7   
Bathurst Manor, Wilson Heights, Downsview North                      23   
Bayview Village                                                       4   
Bedford Park, Lawrence Manor East                                    24   
...                                                                 ...   
Willowdale, Willowdale East                                          34   
Willowdale, Willowdale West                                           6   
Woburn                                                                4   
Woodbine Heights                                                      7   
York Mills West                                                       2   

                                                 Venue  Venue Latitude  \
Neighborhood                                                             
Agincourt                                            4               4   
Alderwood, Long Branch                               7               7   
Bathurst Manor, Wilson Heights, Downsview North     23              23   
Bayview Village                                      4               4   
Bedford Park, Lawrence Manor East                   24              24   
...                                                ...             ...   
Willowdale, Willowdale East                         34              34   
Willowdale, Willowdale West                          6               6   
Woburn                                               4               4   
Woodbine Heights                                     7               7   
York Mills West                                      2               2   

                                                 Venue Longitude  \
Neighborhood                                                       
Agincourt                                                      4   
Alderwood, Long Branch                                         7   
Bathurst Manor, Wilson Heights, Downsview North               23   
Bayview Village                                                4   
Bedford Park, Lawrence Manor East                             24   
...                                                          ...   
Willowdale, Willowdale East                                   34   
Willowdale, Willowdale West                                    6   
Woburn                                                         4   
Woodbine Heights                                               7   
York Mills West                                                2   

                                                 Venue Category  
Neighborhood                                                     
Agincourt                                                     4  
Alderwood, Long Branch                                        7 

In [55]:
toronto_categories = toronto_venues.groupby('Venue Category').count()
toronto_categories.head(60)

Neighborhood  Neighborhood Latitude  \
Venue Category                                                   
Accessories Store                     1                      1   
Airport                               2                      2   
Airport Food Court                    1                      1   
Airport Gate                          1                      1   
Airport Lounge                        2                      2   
Airport Service                       3                      3   
Airport Terminal                      2                      2   
American Restaurant                  26                     26   
Antique Shop                          3                      3   
Aquarium                              5                      5   
Art Gallery                          13                     13   
Art Museum                            2                      2   
Arts & Crafts Store                   3                      3   
Asian Restaurant                     14                     14   
Athletics & Sports                    5                      5   
Auto Garage                           1                      1   
Auto Workshop                         1                      1   
BBQ Joint                             2                      2   
Baby Store                            1                      1   
Bagel Shop                            4                      4   
Bakery                               46                     46   
Bank                                 27                     27   
Bar                                  28                     28   
Baseball Field                        5                      5   
Baseball Stadium                      2                      2   
Basketball Court                      1                      1   
Basketball Stadium                    4                      4   
Beach                                 1                      1   
Bed & Breakfast                       1                      1   
Beer Bar                             17                     17   
Beer Store                           10                     10   
Belgian Restaurant                    2                      2   
Bike Shop                             2                      2   
Bistro                                4                      4   
Boat or Ferry                         1                      1   
Bookstore                            15                     15   
Boutique                              5                      5   
Brazilian Restaurant                  1                      1   
Breakfast Spot                       21                     21   
Brewery                              13                     13   
Bridal Shop                           1                      1   
Bubble Tea Shop                       9                      9   
Building                              1                      1   
Burger Joint                         14                     14   
Burrito Place                        12                     12   
Bus Line                              5                      5   
Bus Station                           1                      1   
Bus Stop                              1                      1   
Business Service                      2                      2   
Butcher                               3                      3   
Cable Car                             1                      1   
Café                                 92                     92   
Cajun / Creole Restaurant             1                      1   
Candy Store                           1                      1   
Caribbean Restaurant                  8                      8   
Cheese Shop                           6                      6   
Chinese Restaurant                   16                     16   
Chocolate Shop                        3                      3   
Church                                1                      1   
C

In the following two lines we have the list of Chinese restaurants and Asian restaurants in Toronto so we can see what neighborhoods they are in.

In [91]:
chinese_venues=toronto_venues[toronto_venues['Venue Category']=='Chinese Restaurant']
chinese_venues

Neighborhood  \
43    Dorset Park, Wexford Heights, Scarborough Town...   
58                                            Agincourt   
61              Clarks Corners, Tam O'Shanter, Sullivan   
75                        Steeles West, L'Amoreaux West   
148                        Fairview, Henry Farm, Oriole   
160                                     Bayview Village   
233                                           Don Mills   
262     Bathurst Manor, Wilson Heights, Downsview North   
482                   North Toronto West, Lawrence Park   
584                         St. James Town, Cabbagetown   
589                         St. James Town, Cabbagetown   
796                            Garden District, Ryerson   
1003                                 Central Bay Street   
1144  Harbourfront East, Union Station, Toronto Islands   
1292           Toronto Dominion Centre, Design Exchange   
2093                                          Westmount   

      Neighborhood Latitude  Neighborhood Longitude  \
43                43.757410              -79.273304   
58                43.794200              -79.262029   
61                43.781638              -79.304302   
75                43.799525              -79.318389   
148               43.778517              -79.346556   
160               43.786947              -79.385975   
233               43.725900              -79.340923   
262               43.754328              -79.442259   
482               43.715383              -79.405678   
584               43.667967              -79.367675   
589               43.667967              -79.367675   
796               43.657162              -79.378937   
1003              43.657952              -79.387383   
1144              43.640816              -79.381752   
1292              43.647177              -79.381576   
2093              43.696319              -79.532242   

                                   Venue  Venue Latitude  Venue Longitude  \
43                    Kim Kim restaurant       43.753833       -79.276611   
58      Royal Chinese Seafood Restaurant       43.798496       -79.262196   
61    The Royal Chinese Restaurant 避風塘小炒       43.780505       -79.298844   
75         Mr Congee Chinese Cuisine 龍粥記       43.798879       -79.318335   
148                           Manchu Wok       43.778225       -79.343302   
160        Sun Star Chinese Cuisine 翠景小炒       43.787914       -79.381234   
233                     Congee Star 帝王名粥       43.726586       -79.341833   
262                          China Court       43.755780       -79.437437   
482                            C'est Bon       43.716785       -79.400406   
584                        China Gourmet       43.664180       -79.368359   
589               Tender Trap Restaurant       43.667724       -79.369485   
796               GB Hand-Pulled Noodles       43.656434       -79.383783   
1003                                Duck       43.660739       -79.383913   
1144                  Pearl Harbourfront       43.638157       -79.380688   
1292                    Szechuan Express       43.646973       -79.379549   
2093              Mayflower Chinese Food       43.692753       -79.531566   

          Venue Category  
43    Chinese Restaurant  
58    Chinese Restaurant  
61    Chinese Restaurant  
75    Chinese Restaurant  
148   Chinese Restaurant  
160   Chinese Restaurant  
233   Chinese Restaurant  
262   Chinese Restaurant  
482   Chinese Restaurant  
584   Chinese Restaurant  
589   Chinese Restaurant  
796   Chinese Restaurant  
1003  Chinese Restaurant  
1144  Chinese Restaurant  
1292  Chinese Restaurant  
2093  Chinese Restaurant

In [74]:
asian_venues = toronto_venues[toronto_venues['Venue Category']=='Asian Restaurant']
asian_venues

Neighborhood  Neighborhood Latitude  \
123               Fairview, Henry Farm, Oriole              43.778517   
235                                  Don Mills              43.725900   
712                  Regent Park, Harbourfront              43.654260   
874                             St. James Town              43.651494   
1025                  Richmond, Adelaide, King              43.650571   
1248  Toronto Dominion Centre, Design Exchange              43.647177   
1282  Toronto Dominion Centre, Design Exchange              43.647177   
1380            Commerce Court, Victoria Hotel              43.648198   
1691    First Canadian Place, Underground city              43.648429   
1728    First Canadian Place, Underground city              43.648429   
1744    First Canadian Place, Underground city              43.648429   
1776                                 Glencairn              43.709577   
1822                  Little Portugal, Trinity              43.647927   
1844                  Little Portugal, Trinity              43.647927   

      Neighborhood Longitude                Venue  Venue Latitude  \
123               -79.346556         Spring Rolls       43.777351   
235               -79.340923          Pho 88 發發餐廳       43.726642   
712               -79.360636                Izumi       43.649970   
874               -79.375418           Sweet Lulu       43.650557   
1025              -79.384568   Bosk at Shangri-La       43.649023   
1248              -79.381576   Bosk at Shangri-La       43.649023   
1282              -79.381576           Sweet Lulu       43.650557   
1380              -79.379817           Sweet Lulu       43.650557   
1691              -79.382280   Bosk at Shangri-La       43.649023   
1728              -79.382280           Sweet Lulu       43.650557   
1744              -79.382280          Noodle King       43.651706   
1776              -79.445073  Li Cheng Restaurant       43.708828   
1822              -79.419750        Foxley Bistro       43.648643   
1844              -79.419750              Hanmoto       43.649505   

      Venue Longitude    Venue Category  
123        -79.343977  Asian Restaurant  
235        -79.342345  Asian Restaurant  
712        -79.360153  Asian Restaurant  
874        -79.381175  Asian Restaurant  
1025       -79.385826  Asian Restaurant  
1248       -79.385826  Asian Restaurant  
1282       -79.381175  Asian Restaurant  
1380       -79.381175  Asian Restaurant  
1691       -79.385826  Asian Restaurant  
1728       -79.381175  Asian Restaurant  
1744       -79.383046  Asian Restaurant  
1776       -79.443366  Asian Restaurant  
1822       -79.420495  Asian Restaurant  
1844       -79.422768  Asian Restaurant

In [101]:
frames = [asian_venues,chinese_venues]
asian_venues_two = pd.concat(frames)
asian_venues_two

Neighborhood  \
123                        Fairview, Henry Farm, Oriole   
235                                           Don Mills   
712                           Regent Park, Harbourfront   
874                                      St. James Town   
1025                           Richmond, Adelaide, King   
1248           Toronto Dominion Centre, Design Exchange   
1282           Toronto Dominion Centre, Design Exchange   
1380                     Commerce Court, Victoria Hotel   
1691             First Canadian Place, Underground city   
1728             First Canadian Place, Underground city   
1744             First Canadian Place, Underground city   
1776                                          Glencairn   
1822                           Little Portugal, Trinity   
1844                           Little Portugal, Trinity   
43    Dorset Park, Wexford Heights, Scarborough Town...   
58                                            Agincourt   
61              Clarks Corners, Tam O'Shanter, Sullivan   
75                        Steeles West, L'Amoreaux West   
148                        Fairview, Henry Farm, Oriole   
160                                     Bayview Village   
233                                           Don Mills   
262     Bathurst Manor, Wilson Heights, Downsview North   
482                   North Toronto West, Lawrence Park   
584                         St. James Town, Cabbagetown   
589                         St. James Town, Cabbagetown   
796                            Garden District, Ryerson   
1003                                 Central Bay Street   
1144  Harbourfront East, Union Station, Toronto Islands   
1292           Toronto Dominion Centre, Design Exchange   
2093                                          Westmount   

      Neighborhood Latitude  Neighborhood Longitude  \
123               43.778517              -79.346556   
235               43.725900              -79.340923   
712               43.654260              -79.360636   
874               43.651494              -79.375418   
1025              43.650571              -79.384568   
1248              43.647177              -79.381576   
1282              43.647177              -79.381576   
1380              43.648198              -79.379817   
1691              43.648429              -79.382280   
1728              43.648429              -79.382280   
1744              43.648429              -79.382280   
1776              43.709577              -79.445073   
1822              43.647927              -79.419750   
1844              43.647927              -79.419750   
43                43.757410              -79.273304   
58                43.794200              -79.262029   
61                43.781638              -79.304302   
75                43.799525              -79.318389   
148               43.778517              -79.346556   
160               43.786947              -79.385975   
233               43.725900              -79.340923   
262               43.754328              -79.442259   
482               43.715383              -79.405678   
584               43.667967              -79.367675   
589               43.667967              -79.367675   
796               43.657162              -79.378937   
1003              43.657952              -79.387383   
1144              43.640816              -79.381752   
1292              43.647177              -79.381576   
2093              43.696319              -79.532242   

                                   Venue  Venue Latitude  Venue Longitude  \
123                         Spring Rolls       43.777351       -79.343977   
235                          Pho 88 發發餐廳       43.726642       -79.342345   
712                                Izumi       43.649970       -79.360153   
874                           Sweet Lulu       43.650557       -79.381175   
1025                  Bosk at Shangri-La       43.649023       -79.385826   
1248                  Bosk at Shangri-La     

Now we will do one hot encoding.

In [114]:
# one hot encoding
toronto_onehot = pd.get_dummies(toronto_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
toronto_onehot['Neighborhood'] = toronto_venues['Neighborhood'] 

# move neighborhood column to the first column
fixed_columns = [toronto_onehot.columns[-1]] + list(toronto_onehot.columns[:-1])
toronto_onehot = toronto_onehot[fixed_columns]

toronto_onehot.head()

Yoga Studio  Accessories Store  Airport  Airport Food Court  Airport Gate  \
0            0                  0        0                   0             0   
1            0                  0        0                   0             0   
2            0                  0        0                   0             0   
3            0                  0        0                   0             0   
4            0                  0        0                   0             0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0               0                0                 0                    0   
1               0                0                 0                    0   
2               0                0                 0                    0   
3               0                0                 0                    0   
4               0                0                 0                    0   

   Antique Shop  ...  Train Station  Turkish Restaurant  \
0             0  ...              0                   0   
1             0  ...              0                   0   
2             0  ...              0                   0   
3             0  ...              0                   0   
4             0  ...              0                   0   

   Vegetarian / Vegan Restaurant  Video Game Store  Vietnamese Restaurant  \
0                              0                 0                      0   
1                              0                 0                      0   
2                              0                 0                      0   
3                              0                 0                      0   
4                              0                 0                      0   

   Warehouse Store  Wine Bar  Wine Shop  Wings Joint  Women's Store  
0                0         0          0            0              0  
1                0         0          0            0              0  
2                0         0          0            0              0  
3                0         0          0            0              0  
4                0         0          0            0              0  

[5 rows x 269 columns]

In [217]:
toronto_grouped = toronto_onehot.groupby('Neighborhood').mean().reset_index()
toronto_grouped

Neighborhood  Yoga Studio  \
0                                         Agincourt          0.0   
1                            Alderwood, Long Branch          0.0   
2   Bathurst Manor, Wilson Heights, Downsview North          0.0   
3                                   Bayview Village          0.0   
4                 Bedford Park, Lawrence Manor East          0.0   
..                                              ...          ...   
91                      Willowdale, Willowdale East          0.0   
92                      Willowdale, Willowdale West          0.0   
93                                           Woburn          0.0   
94                                 Woodbine Heights          0.0   
95                                  York Mills West          0.0   

    Accessories Store  Airport  Airport Food Court  Airport Gate  \
0                 0.0      0.0                 0.0           0.0   
1                 0.0      0.0                 0.0           0.0   
2                 0.0      0.0                 0.0           0.0   
3                 0.0      0.0                 0.0           0.0   
4                 0.0      0.0                 0.0           0.0   
..                ...      ...                 ...           ...   
91                0.0      0.0                 0.0           0.0   
92                0.0      0.0                 0.0           0.0   
93                0.0      0.0                 0.0           0.0   
94                0.0      0.0                 0.0           0.0   
95                0.0      0.0                 0.0           0.0   

    Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0              0.0              0.0               0.0             0.000000   
1              0.0              0.0               0.0             0.000000   
2              0.0              0.0               0.0             0.000000   
3              0.0              0.0               0.0             0.000000   
4              0.0              0.0               0.0             0.041667   
..             ...              ...               ...                  ...   
91             0.0              0.0               0.0             0.000000   
92             0.0              0.0               0.0             0.000000   
93             0.0              0.0               0.0             0.000000   
94             0.0              0.0               0.0             0.000000   
95             0.0              0.0               0.0             0.000000   

    ...  Train Station  Turkish Restaurant  Vegetarian / Vegan Restaurant  \
0   ...            0.0                 0.0                            0.0   
1   ...            0.0                 0.0                            0.0   
2   ...            0.0                 0.0                            0.0   
3   ...            0.0                 0.0                            0.0   
4   ...            0.0                 0.0                            0.0   
..  ...            ...                 ...                            ...   
91  ...            0.0                 0.0                            0.0   
92  ...            0.0                 0.0                            0.0   
93  ...            0.0                 0.0                            0.0   
94  ...            0.0                 0.0                            0.0   
95  ...            0.0                 0.0                            0.0   

    Video Game Store  Vietnamese Restaurant  Warehouse Store  Wine Bar  \
0                0.0               0.000000              0.0       0.0   
1                0.0               0.000000              0.0       0.0   
2                0.0               0.000000              0.0       0.0   
3                0.0               0.000000              0.0       0.0   
4                0.0               0.000000              0.0       0.0   
..               ...                    ...              ...       ...   
91               0.0     

In [218]:
# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans# set number of clusters
kclusters = 20

toronto_grouped_clustering = toronto_grouped.drop('Neighborhood', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(toronto_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:20] 

array([19,  2, 10, 10, 10, 10, 10, 10,  1, 10,  9, 13, 10, 10, 10, 10, 10,
        0, 10, 10], dtype=int32)

In [219]:
toronto_grouped.insert(0, 'Cluster Labels', kmeans.labels_)


As you can see we clustered the neighborhoods into 20 different clusters.  All but two of the Chinese and Asian restaurants are in group #10 (which as you will see below has 51 members).  

In [220]:
toronto_grouped[toronto_grouped['Asian Restaurant']>0]

Cluster Labels                              Neighborhood  Yoga Studio  \
18              10            Commerce Court, Victoria Hotel     0.000000   
22              10                                 Don Mills     0.000000   
28              10              Fairview, Henry Farm, Oriole     0.000000   
29              10    First Canadian Place, Underground city     0.000000   
32              10                                 Glencairn     0.000000   
48              10                  Little Portugal, Trinity     0.023256   
63              10                 Regent Park, Harbourfront     0.022727   
64              10                  Richmond, Adelaide, King     0.000000   
72              10                            St. James Town     0.000000   
83              10  Toronto Dominion Centre, Design Exchange     0.000000   

    Accessories Store  Airport  Airport Food Court  Airport Gate  \
18                0.0      0.0                 0.0           0.0   
22                0.0      0.0                 0.0           0.0   
28                0.0      0.0                 0.0           0.0   
29                0.0      0.0                 0.0           0.0   
32                0.0      0.0                 0.0           0.0   
48                0.0      0.0                 0.0           0.0   
63                0.0      0.0                 0.0           0.0   
64                0.0      0.0                 0.0           0.0   
72                0.0      0.0                 0.0           0.0   
83                0.0      0.0                 0.0           0.0   

    Airport Lounge  Airport Service  Airport Terminal  ...  Train Station  \
18             0.0              0.0               0.0  ...           0.00   
22             0.0              0.0               0.0  ...           0.00   
28             0.0              0.0               0.0  ...           0.00   
29             0.0              0.0               0.0  ...           0.01   
32             0.0              0.0               0.0  ...           0.00   
48             0.0              0.0               0.0  ...           0.00   
63             0.0              0.0               0.0  ...           0.00   
64             0.0              0.0               0.0  ...           0.00   
72             0.0              0.0               0.0  ...           0.00   
83             0.0              0.0               0.0  ...           0.01   

    Turkish Restaurant  Vegetarian / Vegan Restaurant  Video Game Store  \
18                 0.0                       0.020000          0.000000   
22                 0.0                       0.000000          0.000000   
28                 0.0                       0.000000          0.014706   
29                 0.0                       0.010000          0.000000   
32                 0.0                       0.000000          0.000000   
48                 0.0                       0.046512          0.000000   
63                 0.0                       0.000000          0.000000   
64                 0.0                       0.010870          0.000000   
72                 0.0                       0.012658          0.000000   
83                 0.0                       0.010000          0.000000   

    Vietnamese Restaurant  Warehouse Store  Wine Bar  Wine Shop  Wings Joint  \
18               0.000000              0.0  0.010000   0.000000          0.0   
22               0.000000              0.0  0.000000   0.000000          0.0   
28               0.000000              0.0  0.000000   0.000000          0.0   
29               0.000000              0.0  0.010000   0.000000          0.0   
32               0.000000              0.0  0.000000   0.000000          0.0   
48               0.023256              0.0  0.023256   0.000000          0.0   
63               0.000000              0.0  0.000000   0.022727          0.0   
64               0.000000              0.0  0.000000   0.000000          0.0   
72           

In [221]:
toronto_grouped[toronto_grouped['Chinese Restaurant']>0]

Cluster Labels                                       Neighborhood  \
0               19                                          Agincourt   
2               10    Bathurst Manor, Wilson Heights, Downsview North   
3               10                                    Bayview Village   
13              10                                 Central Bay Street   
16              10            Clarks Corners, Tam O'Shanter, Sullivan   
22              10                                          Don Mills   
23              10  Dorset Park, Wexford Heights, Scarborough Town...   
28              10                       Fairview, Henry Farm, Oriole   
31              10                           Garden District, Ryerson   
35              10  Harbourfront East, Union Station, Toronto Islands   
55              10                  North Toronto West, Lawrence Park   
73              10                        St. James Town, Cabbagetown   
74              10                      Steeles West, L'Amoreaux West   
83              10           Toronto Dominion Centre, Design Exchange   
87               2                                          Westmount   

    Yoga Studio  Accessories Store  Airport  Airport Food Court  Airport Gate  \
0      0.000000                0.0      0.0                 0.0           0.0   
2      0.000000                0.0      0.0                 0.0           0.0   
3      0.000000                0.0      0.0                 0.0           0.0   
13     0.016667                0.0      0.0                 0.0           0.0   
16     0.000000                0.0      0.0                 0.0           0.0   
22     0.000000                0.0      0.0                 0.0           0.0   
23     0.000000                0.0      0.0                 0.0           0.0   
28     0.000000                0.0      0.0                 0.0           0.0   
31     0.000000                0.0      0.0                 0.0           0.0   
35     0.000000                0.0      0.0                 0.0           0.0   
55     0.050000                0.0      0.0                 0.0           0.0   
73     0.000000                0.0      0.0                 0.0           0.0   
74     0.000000                0.0      0.0                 0.0           0.0   
83     0.000000                0.0      0.0                 0.0           0.0   
87     0.000000                0.0      0.0                 0.0           0.0   

    Airport Lounge  Airport Service  Airport Terminal  ...  Train Station  \
0              0.0              0.0               0.0  ...           0.00   
2              0.0              0.0               0.0  ...           0.00   
3              0.0              0.0               0.0  ...           0.00   
13             0.0              0.0               0.0  ...           0.00   
16             0.0              0.0               0.0  ...           0.00   
22             0.0              0.0               0.0  ...           0.00   
23             0.0              0.0               0.0  ...           0.00   
28             0.0              0.0               0.0  ...           0.00   
31             0.0              0.0               0.0  ...           0.00   
35             0.0              0.0               0.0  ...           0.01   
55             0.0              0.0               0.0  ...           0.00   
73             0.0              0.0               0.0  ...           0.00   
74             0.0              0.0               0.0  ...           0.00   
83             0.0              0.0               0.0  ...           0.01   
87             0.0              0.0               0.0  ...           0.00   

    Turkish Restaurant  Vegetarian / Vegan Restaurant  Video Game Store  \
0                  0.0                       0.000000          0.000000   
2                  0.0                       0.000000          0.000000   
3                  0.0                       0.000000          0.000000   
13   

In [222]:
cluster_count = toronto_grouped.groupby('Cluster Labels').count()
cluster_count

Neighborhood  Yoga Studio  Accessories Store  Airport  \
Cluster Labels                                                          
0                          1            1                  1        1   
1                          8            8                  8        8   
2                          8            8                  8        8   
3                          1            1                  1        1   
4                          2            2                  2        2   
5                          2            2                  2        2   
6                          3            3                  3        3   
7                          1            1                  1        1   
8                          1            1                  1        1   
9                          2            2                  2        2   
10                        51           51                 51       51   
11                         1            1                  1        1   
12                         3            3                  3        3   
13                         6            6                  6        6   
14                         1            1                  1        1   
15                         1            1                  1        1   
16                         1            1                  1        1   
17                         1            1                  1        1   
18                         1            1                  1        1   
19                         1            1                  1        1   

                Airport Food Court  Airport Gate  Airport Lounge  \
Cluster Labels                                                     
0                                1             1               1   
1                                8             8               8   
2                                8             8               8   
3                                1             1               1   
4                                2             2               2   
5                                2             2               2   
6                                3             3               3   
7                                1             1               1   
8                                1             1               1   
9                                2             2               2   
10                              51            51              51   
11                               1             1               1   
12                               3             3               3   
13                               6             6               6   
14                               1             1               1   
15                               1             1               1   
16                               1             1               1   
17                               1             1               1   
18                               1             1               1   
19                               1             1               1   

                Airport Service  Airport Terminal  American Restaurant  ...  \
Cluster Labels                                                          ...   
0                             1                 1                    1  ...   
1                             8                 8                    8  ...   
2                             8                 8                    8  ...   
3                             1                 1                    1  ...   
4                             2                 2                    2  ...   
5                             2                 2                    2  ...   
6                             3                 3                    3  ...   
7                             1                 1                    1  ...   
8                             1                 1                    1  ...   
9                             2            

In [225]:
big_toronto_cluster = toronto_grouped[toronto_grouped['Cluster Labels']==10]
big_toronto_cluster

Cluster Labels                                       Neighborhood  \
2               10    Bathurst Manor, Wilson Heights, Downsview North   
3               10                                    Bayview Village   
4               10                  Bedford Park, Lawrence Manor East   
5               10                                        Berczy Park   
6               10                        Birch Cliff, Cliffside West   
7               10       Brockton, Parkdale Village, Exhibition Place   
9               10  CN Tower, King and Spadina, Railway Lands, Har...   
12              10                                          Cedarbrae   
13              10                                 Central Bay Street   
14              10                                           Christie   
15              10                               Church and Wellesley   
16              10            Clarks Corners, Tam O'Shanter, Sullivan   
18              10                     Commerce Court, Victoria Hotel   
19              10                                         Davisville   
22              10                                          Don Mills   
23              10  Dorset Park, Wexford Heights, Scarborough Town...   
24              10                                          Downsview   
25              10                       Dufferin, Dovercourt Village   
27              10  Eringate, Bloordale Gardens, Old Burnhamthorpe...   
28              10                       Fairview, Henry Farm, Oriole   
29              10             First Canadian Place, Underground city   
31              10                           Garden District, Ryerson   
32              10                                          Glencairn   
34              10                  Guildwood, Morningside, West Hill   
35              10  Harbourfront East, Union Station, Toronto Islands   
36              10                      High Park, The Junction South   
43              10          Kensington Market, Chinatown, Grange Park   
45              10                   Lawrence Manor, Lawrence Heights   
47              10                                            Leaside   
48              10                           Little Portugal, Trinity   
51              10  Mimico NW, The Queensway West, South of Bloor,...   
53              10       New Toronto, Mimico South, Humber Bay Shores   
55              10                  North Toronto West, Lawrence Park   
59              10                             Parkdale, Roncesvalles   
63              10                          Regent Park, Harbourfront   
64              10                           Richmond, Adelaide, King   
68              10                                 Runnymede, Swansea   
72              10                                     St. James Town   
73              10                        St. James Town, Cabbagetown   
74              10                      Steeles West, L'Amoreaux West   
75              10                                     Stn A PO Boxes   
76              10                                    Studio District   
77              10  Summerhill West, Rathnelly, South Hill, Forest...   
78              10                The Annex, North Midtown, Yorkville   
80              10                       The Danforth West, Riverdale   
82              10                                   Thorncliffe Park   
83              10           Toronto Dominion Centre, Design Exchange   
84              10                     University of Toronto, Harbord   
89              10                                  Wexford, Maryvale   
91              10                        Willowdale, Willowdale East   
94              10                                   Woodbine Heights   

    Yoga Studio  Accessories Store   Airport  Airport Food Court  \
2      0.000000           0.000000  0.000000            0.000000   
3      0.000000           0.000000  0.000000            0.000000   
4  

In [235]:
cluster_notasian = big_toronto_cluster[big_toronto_cluster['Asian Restaurant']==0]
cluster_notasian[['Cluster Labels','Neighborhood','Asian Restaurant','Chinese Restaurant']]

Cluster Labels                                       Neighborhood  \
2               10    Bathurst Manor, Wilson Heights, Downsview North   
3               10                                    Bayview Village   
4               10                  Bedford Park, Lawrence Manor East   
5               10                                        Berczy Park   
6               10                        Birch Cliff, Cliffside West   
7               10       Brockton, Parkdale Village, Exhibition Place   
9               10  CN Tower, King and Spadina, Railway Lands, Har...   
12              10                                          Cedarbrae   
13              10                                 Central Bay Street   
14              10                                           Christie   
15              10                               Church and Wellesley   
16              10            Clarks Corners, Tam O'Shanter, Sullivan   
19              10                                         Davisville   
23              10  Dorset Park, Wexford Heights, Scarborough Town...   
24              10                                          Downsview   
25              10                       Dufferin, Dovercourt Village   
27              10  Eringate, Bloordale Gardens, Old Burnhamthorpe...   
31              10                           Garden District, Ryerson   
34              10                  Guildwood, Morningside, West Hill   
35              10  Harbourfront East, Union Station, Toronto Islands   
36              10                      High Park, The Junction South   
43              10          Kensington Market, Chinatown, Grange Park   
45              10                   Lawrence Manor, Lawrence Heights   
47              10                                            Leaside   
51              10  Mimico NW, The Queensway West, South of Bloor,...   
53              10       New Toronto, Mimico South, Humber Bay Shores   
55              10                  North Toronto West, Lawrence Park   
59              10                             Parkdale, Roncesvalles   
68              10                                 Runnymede, Swansea   
73              10                        St. James Town, Cabbagetown   
74              10                      Steeles West, L'Amoreaux West   
75              10                                     Stn A PO Boxes   
76              10                                    Studio District   
77              10  Summerhill West, Rathnelly, South Hill, Forest...   
78              10                The Annex, North Midtown, Yorkville   
80              10                       The Danforth West, Riverdale   
82              10                                   Thorncliffe Park   
84              10                     University of Toronto, Harbord   
89              10                                  Wexford, Maryvale   
91              10                        Willowdale, Willowdale East   
94              10                                   Woodbine Heights   

    Asian Restaurant  Chinese Restaurant  
2                0.0            0.043478  
3                0.0            0.250000  
4                0.0            0.000000  
5                0.0            0.000000  
6                0.0            0.000000  
7                0.0            0.000000  
9                0.0            0.000000  
12               0.0            0.000000  
13               0.0            0.016667  
14               0.0            0.000000  
15               0.0            0.000000  
16               0.0            0.076923  
19               0.0            0.000000  
23               0.0            0.166667  
24               0.0            0.000000  
25               0.0            0.000000  
27               0.0            0.000000  
31               0.0            0.010000  
34               0.0            0.000000  
35               0.0            0.010000  
36               0.0            0.

In [236]:
cluster_notchinese = big_toronto_cluster[big_toronto_cluster['Chinese Restaurant']==0]
cluster_notchinese[['Cluster Labels','Neighborhood','Asian Restaurant','Chinese Restaurant']]

Cluster Labels                                       Neighborhood  \
4               10                  Bedford Park, Lawrence Manor East   
5               10                                        Berczy Park   
6               10                        Birch Cliff, Cliffside West   
7               10       Brockton, Parkdale Village, Exhibition Place   
9               10  CN Tower, King and Spadina, Railway Lands, Har...   
12              10                                          Cedarbrae   
14              10                                           Christie   
15              10                               Church and Wellesley   
18              10                     Commerce Court, Victoria Hotel   
19              10                                         Davisville   
24              10                                          Downsview   
25              10                       Dufferin, Dovercourt Village   
27              10  Eringate, Bloordale Gardens, Old Burnhamthorpe...   
29              10             First Canadian Place, Underground city   
32              10                                          Glencairn   
34              10                  Guildwood, Morningside, West Hill   
36              10                      High Park, The Junction South   
43              10          Kensington Market, Chinatown, Grange Park   
45              10                   Lawrence Manor, Lawrence Heights   
47              10                                            Leaside   
48              10                           Little Portugal, Trinity   
51              10  Mimico NW, The Queensway West, South of Bloor,...   
53              10       New Toronto, Mimico South, Humber Bay Shores   
59              10                             Parkdale, Roncesvalles   
63              10                          Regent Park, Harbourfront   
64              10                           Richmond, Adelaide, King   
68              10                                 Runnymede, Swansea   
72              10                                     St. James Town   
75              10                                     Stn A PO Boxes   
76              10                                    Studio District   
77              10  Summerhill West, Rathnelly, South Hill, Forest...   
78              10                The Annex, North Midtown, Yorkville   
80              10                       The Danforth West, Riverdale   
82              10                                   Thorncliffe Park   
84              10                     University of Toronto, Harbord   
89              10                                  Wexford, Maryvale   
91              10                        Willowdale, Willowdale East   
94              10                                   Woodbine Heights   

    Asian Restaurant  Chinese Restaurant  
4           0.000000                 0.0  
5           0.000000                 0.0  
6           0.000000                 0.0  
7           0.000000                 0.0  
9           0.000000                 0.0  
12          0.000000                 0.0  
14          0.000000                 0.0  
15          0.000000                 0.0  
18          0.010000                 0.0  
19          0.000000                 0.0  
24          0.000000                 0.0  
25          0.000000                 0.0  
27          0.000000                 0.0  
29          0.030000                 0.0  
32          0.200000                 0.0  
34          0.000000                 0.0  
36          0.000000                 0.0  
43          0.000000                 0.0  
45          0.000000                 0.0  
47          0.000000                 0.0  
48          0.046512                 0.0  
51          0.000000                 0.0  
53          0.000000                 0.0  
59          0.000000                 0.0  
63          0.022727                 0.0  
64          0.010870                 0